modules/memfault-firmware-sdk/memfault_lte_coredump.c

```diff
@@ -181,48 +179,54 @@ static void on_modem_lib_shutdown(void *ctx)
 
 static void lte_handler(const struct lte_lc_evt *const evt)
 {
-	if (evt->type == LTE_LC_EVT_NW_REG_STATUS) {
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS:
 		switch (evt->nw_reg_status) {
 		case LTE_LC_NW_REG_REGISTERED_HOME:
 			__fallthrough;
 		case LTE_LC_NW_REG_REGISTERED_ROAMING:
 			event_send(EVENT_LTE_REGISTERED);
+
 			break;
 		case LTE_LC_NW_REG_UNKNOWN:
 			__fallthrough;
 		case LTE_LC_NW_REG_UICC_FAIL:
 			__fallthrough;
 		case LTE_LC_NW_REG_NOT_REGISTERED:
-			 __fallthrough;
+			__fallthrough;
 		case LTE_LC_NW_REG_REGISTRATION_DENIED:
 			event_send(EVENT_LTE_DEREGISTERED);
+
+			break;
 		default:
-			/* Don't care */
 			break;
 		}
-	}
-}
-
-static void pdn_event_handler(uint8_t cid, enum pdn_event event, int reason)
-{
-	switch (event) {
-	case PDN_EVENT_ACTIVATED:
-		if (cid == 0) {
-			event_send(EVENT_PDN_ACTIVATED);
-		}
 
 		break;
-	case PDN_EVENT_DEACTIVATED:
-		if (cid == 0) {
+	case LTE_LC_EVT_PDN:
+		switch (evt->pdn.type) {
+		case LTE_LC_EVT_PDN_ACTIVATED:
+			if (evt->pdn.cid == 0) {
+				event_send(EVENT_PDN_ACTIVATED);
+			}
+
+			break;
+		case LTE_LC_EVT_PDN_DEACTIVATED:
+			if (evt->pdn.cid == 0) {
+				event_send(EVENT_PDN_DEACTIVATED);
+			}
+
+			break;
+		case LTE_LC_EVT_PDN_NETWORK_DETACH:
 			event_send(EVENT_PDN_DEACTIVATED);
```

Comment:
The `PDN_EVENT_NETWORK_DETACH` event is sent for all cid's by the PDN library. The PDN implementation in the LTE LC is different, it sends the `LTE_LC_EVT_PDN_NETWORK_DETACH` event only once, but with the cellular profile ID. This may need changes in the application, so this needs to be documented in the migration guide, I think?

Reply:
Yeah, it's probably best that way. 
I have added that behavior now in bc596c9af8b8dc4ec52263d1e926e96a63e09e5d. 
Please have a look if this is what you also wanted, @lucasvdp